### PR TITLE
dreport: added new plugins to collect additional data for BMC dump

### PR DESCRIPTION
**Pull request by devenrao**
Pulled the following commits from master and also changed to use ZSTD compression in downstream

https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52899
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52913
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52891
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52849
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52944
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53082
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52594
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52943
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52852
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52792
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/52850

### CODE REVIEW

File: tools/dreport.d/plugins.d/inventory
Comment (RameshIyyar) [inline tools/dreport.d/plugins.d/inventory:20]: it is separate change, please push a separate commit

File: tools/dreport.d/ibm.d/package
Comment (RameshIyyar) [inline tools/dreport.d/ibm.d/package:7]: the dump size is increasing but, I think this plugin is called only for the [p10](https://github.ibm.com/openbmc/openbmc/pull/2267/commits/0722b480c16201ed2e535f519768d282d2651a82#diff-b9dcea4197aa53ae9ebac1f8029a44885ac99a9bb695c85b8f0086771e766d9fR57) and [witherspoon tacoma](https://github.ibm.com/openbmc/openbmc/pull/2267/commits/0722b480c16201ed2e535f519768d282d2651a82#diff-b9dcea4197aa53ae9ebac1f8029a44885ac99a9bb695c85b8f0086771e766d9fR56) so I think we are good since we have more flash - is it correct?

**Review by RameshIyyar — COMMENTED** [review]
No functional impacts for downstream, just minor comments and posted in the Gerrit link for the same.

File: tools/dreport.d/ibm.d/package
Comment (devenrao) [reply tools/dreport.d/ibm.d/package:7]: yes only p10

File: tools/dreport.d/plugins.d/inventory
Comment (devenrao) [reply tools/dreport.d/plugins.d/inventory:20]: this changed is added as part of the script so any code commet I should be doing in the same commit

**Comment by ojayanth** [comment]
Can you run  dreport -v and make sure no errors reporting, before merge.

File: tools/dreport.d/plugins.d/inventory
Comment (RameshIyyar) [reply tools/dreport.d/plugins.d/inventory:20]: I did not get it, it is a separate script but, I will leave it to you - just suggestion comment

**Comment by RameshIyyar** [comment]
Looks, like the last commit, is not required, right? did not see any changes

**Comment by ojayanth** [comment]
ZSTD enablement : Cleaner approach will be providing option in dreport  , and bazsed on the  compression type enable the compression type , for IBM code ( ghe)you can pass zstd type. That is celaner way.  The code added here seems to not cleaner.

**Comment by devenrao** [comment]
> ZSTD enablement : Cleaner approach will be providing option in dreport , and bazsed on the compression type enable the compression type , for IBM code ( ghe)you can pass zstd type. That is celaner way. The code added here seems to not cleaner.

agree for master we need to provide option

File: tools/dreport.d/plugins.d/inventory
Comment (devenrao) [reply tools/dreport.d/plugins.d/inventory:20]: agree ramesh sorry this github is bit confusing, i will see that code goes back to original place

**Review by RameshIyyar — APPROVED** [review]
LGTM, there are some suggestions for the upstream but, no functional impact on p10 systems so, I think we are good.

Please execute `dreport` and make sure no errors before merging.

**Comment by ojayanth** [comment]
Devender: can you add description on Known , warnings/errors ( if any with this PR).  Also need Matt's input on esel support.

**Comment by devenrao** [comment]
At present following warnings are dispalyed the same will be fixed in a new patch set, as this patchset has over 13 commits we want this to goahead, and these warnings are not critical.
Thu Apr 21 12:54:11 UTC 2022 ERROR: Failed to collect Hardware dumps
Thu Apr 21 12:54:11 UTC 2022 ERROR: Failed to collect SBE dumps

Regarding below warning
"Thu Apr 21 12:54:11 UTC 2022 ERROR: Failed to collect Attribute list"
corresponding functionality is removed for now and will be added after fixing the attributes command or plugin

**Comment by devenrao** [comment]
ibm_elogall  script is removed based on @spinler approval, matt said "don;t need ibm_elogall anymore, this was just for witherspoon, I am ok if you remove from witherspoon as well, since we aren;t shipping out of master"

**Review by ojayanth — APPROVED** [review]

**Pull request merged**
